src/account.c

```diff
@@ -95,7 +95,11 @@ send_response_in_thread_func (GTask        *task,
       g_autofree char *ruri = NULL;
       g_autoptr(GError) error = NULL;
 
-      ruri = register_document (image, xdp_app_info_get_id (request->app_info), FALSE, FALSE, FALSE, &error);
+      if (xdp_app_info_is_host (request->app_info))
+        ruri = g_strdup (image);
+      else
+        ruri = register_document (image, xdp_app_info_get_id (request->app_info), FALSE, FALSE, FALSE, &error);
+
```

Comment:
I wonder if `register_document()` should receive `XdpAppInfo`, and implement this host/sandbox behaviour internally?

Reply:
The reason I didn't do that is there is no `XdpAppInfo` available in `open-uri.c`; we only have the app ID in that case.